tests/run-make/crate-hash-rustc-version/rmake.rs

```diff
@@ -0,0 +1,46 @@
+// Ensure that crates compiled with different rustc versions cannot
+// be dynamically linked.
+
+//@ ignore-cross-compile
+
+use run_make_support::{cmd, diff, dynamic_lib_name, is_darwin, run, run_fail, rustc};
+
+fn main() {
+    let flags = ["-Cprefer-dynamic", "-Csymbol-mangling-version=v0"];
+    let nm_flag = if is_darwin() { ["-D"].as_slice() } else { [].as_slice() };
+
+    // a.rs is compiled to a dylib
+    rustc().input("a.rs").crate_type("dylib").args(&flags).run();
+
+    // Store symbols
+    let symbols_before = cmd("nm").args(nm_flag).arg(dynamic_lib_name("a")).run().stdout_utf8();
```

Comment:
I've been trying to see if it's possible to remove dependency on `nm` throughout the test suite. Can you check if this works with `llvm_readobj()`? I believe you'd do it like this:

`llvm_readobj().arg("--symbols").input(dynamic_lib_name("a")).run().stdout_utf8()`